src/pygambit/tests/test_node.py

```diff
@@ -213,3 +219,66 @@ def test_node_move_across_games(self):
                           self.game.root, self.extensive_game.root)
         self.assertRaises(pygambit.MismatchError, self.game.move_tree,
                           self.extensive_game.root, self.game.root)
+
+    def test_en_passant_invalid_node(self):
+        """Test the case where we input the same valid nodes in append_move.
+        As the first is resolves the second becomes invalid. Then check if
+        the append_move added anything to the game.
+        """
+        nodes_prior = self.basic_game.nodes()
+        self.assertRaises(
+            pygambit.UndefinedOperationError,
+            self.basic_game.append_move,
+            [self.basic_game.root.children[0],
+             self.basic_game.root.children[0]],
+            "Player 3",
+            ["B", "F"]
+        )
+        nodes_posterior = self.basic_game.nodes()
+
+        assert nodes_posterior == nodes_prior
```

Comment:
This technically doesn't check whether *any* changes have been made to the game.  A convenient way to check this would be to serialise it to the `.efg` format and compare that as a "canonical" representation.